nes/src/cpu/instruction/and.rs

```diff
@@ -1,289 +1,93 @@
-use cpu::{instruction::Execute, Core};
-use memory::ReadAddr;
+use cpu::{instruction::Instruction, operation::Operation, Core};
 
-/// AND memory with accumulator immediate
+/// AND operand with accumulator
 ///
 /// Flags affected: N, Z
-#[derive(Execute)]
-#[opcode = 0x29]
-#[cycles = 2]
-pub struct Immediate;
-
 #[inline(always)]
-fn immediate<T: ReadAddr>(core: &mut Core, memory: &mut T) {
-    core.reg.acc &= core.immediate_addr(memory);
-    update_flags(core);
+fn and(core: &mut Core, operand: u8) {
+    core.reg.acc &= operand;
+    core.reg.status.set_negative(core.reg.acc);
+    core.reg.status.set_zero(core.reg.acc);
 }
 
-/// AND memory with accumulator zero page
+/// AND memory with accumulator immediate
 ///
 /// Flags affected: N, Z
-#[derive(Execute)]
-#[opcode = 0x25]
-#[cycles = 3]
-pub struct ZeroPage;
+pub const IMMEDIATE: Instruction = Instruction {
```

Comment:
I think this would be 💯 if it could be turned into a macro (not a procedural macro just a regular one).

Thinking something like

```rust
instr!(&and, Operation::Immediate, 0x29, 2)
```

And if it has the extra cycle

```rust
instr!(&and, Operation::Immediate, 0x29, 2, true)
```

Reply:
Could we just do that with a constructor for the `Instruction` struct?

Reply:
Yep most definitely :+1: